app/lib/three_scale/analytics/user_tracking.rb

```diff
@@ -30,12 +30,20 @@ def method_missing(method_sym, *args, &block)
           end
         end
 
-        def segment_configured?
-          defined?(::Segment)
+        def self.segment_configured?
+          Features::SegmentConfig.enabled? && defined?(::Segment)
         end
 
         def initialize(config = {})
-          @adapter = segment_configured? ? ::Segment::Analytics.new(config) : NullAdapter.new
+          @config = config
+        end
+
+        private
+
+        attr_reader :config
+
+        def adapter
+          @adapter ||= TrackingAdapter.segment_configured? ? ::Segment::Analytics.new(config) : NullAdapter.new
```

Comment:
Better using `self.class` than `TrackingAdapter`, or you can
```ruby
delegate :segment_configured?, to: :class
```
and call it in the instance.

Reply:
I agree so we can override it by a subclass if ever needed 😄